src/commons_codec/transform/dynamodb.py

```diff
@@ -173,8 +173,24 @@ def values_to_update(self, keys: t.Dict[str, t.Dict[str, str]]) -> str:
             elif isinstance(key_value, str):
                 key_value = "'" + str(key_value).replace("'", "''") + "'"
 
-            if isinstance(key_value, dict):
-                key_value = repr(json.dumps(key_value))
+            elif isinstance(key_value, dict):
+                # TODO: Does it also need escaping of inner TEXT values, like the above?
+                key_value = "'" + json.dumps(key_value) + "'::OBJECT"
+
+            # TODO: ARRAY types do not support JSON syntax yet.
+            #       Let's relay them 1:1, which works for primitive inner types,
+            #       but complex ones need special treatment, like representing
+            #       OBJECTs in CrateDB-native syntax.
+            # FIXME: Find a way to use a custom JSONEncoder for that, in order to
+            #        fully support also nested elements of those.
+            # https://github.com/crate/commons-codec/issues/33
+            elif isinstance(key_value, list):
+                if key_value:
+                    if isinstance(key_value[0], dict):
+                        items = []
+                        for item in key_value:
+                            items.append("{" + ", ".join(f"{key}='{value}'" for key, value in item.items()) + "}")
+                        key_value = "[" + ",".join(items) + "]"
```

Comment:
What is about the other cases where the array element isn't an dict? I

Reply:
They absolutely need to be taken care of, right. Currently, as far as I can see through knowing about data payloads at hand, it might not be of current concerns yet.